torchmetrics/functional/text/squad.py

```diff
@@ -0,0 +1,292 @@
+    """Calculate maximum score for a predicted answer with all reference answers."""
+    scores_for_ground_truths: List[Tensor] = []
+    for ground_truth in ground_truths:
+        score: Tensor = metric_fn(prediction, ground_truth)
+        scores_for_ground_truths.append(score)
+    return torch.max(tensor(scores_for_ground_truths))
+
+
+def _squad_update(
+    preds: Dict[str, str], targets: List[Dict[str, List[Dict[str, List[Dict[str, Union[str, List[Dict[str, str]]]]]]]]]
+) -> Tuple[Tensor, Tensor, Tensor]:
+    """Compute F1 Score and Exact Match for a collection of predictions and references.
+
+    Args:
+        preds:
+            A dictionary mapping an `id` to the predicted `answer`.
+        targets:
+            A list of dictionary mapping `paragraphs` to list of dictionary mapping `qas` to a list of dictionary
+            containing `id` and list of all possible `answers`.
+
+    Return:
+        Tuple containing F1 score, Exact match score and total number of examples.
+
+    Example:
+        >>> from torchmetrics.functional.text.squad import _squad_update
+        >>> predictions = [{"prediction_text": "1976", "id": "56e10a3be3433e1400422b22"}]
+        >>> targets = [{"answers": {"answer_start": [97], "text": ["1976"]}, "id": "56e10a3be3433e1400422b22"}]
+        >>> preds_dict = {prediction["id"]: prediction["prediction_text"] for prediction in predictions}
+        >>> targets_dict = [
+        ...     {
+        ...         "paragraphs": [
+        ...             {
+        ...                 "qas": [
+        ...                     {
+        ...                         "answers": [{"text": answer_text} for answer_text in target["answers"]["text"]],
+        ...                         "id": target["id"],
+        ...                     }
+        ...                     for target in targets
+        ...                 ]
+        ...             }
+        ...         ]
+        ...     }
+        ... ]
+        >>> _squad_update(preds_dict, targets_dict)
+        (tensor(1.), tensor(1.), tensor(1))
+    """
+    f1: Tensor = tensor(0.0)
+    exact_match: Tensor = tensor(0.0)
+    total: Tensor = tensor(0)
+    for article in targets:
+        for paragraph in article["paragraphs"]:
+            for qa in paragraph["qas"]:
+                total += 1
+                if qa["id"] not in preds:
+                    rank_zero_warn(f"Unanswered question {qa['id']} will receive score 0.")
+                    continue
+                ground_truths = list(map(lambda x: x["text"], qa["answers"]))
+                prediction = preds[qa["id"]]
+                exact_match += metric_max_over_ground_truths(compute_exact_match_score, prediction, ground_truths)
+                f1 += metric_max_over_ground_truths(compute_f1_score, prediction, ground_truths)
+
+    return f1, exact_match, total
+
+
+def _squad_compute(scores: Tuple[Tensor, Tensor, Tensor]) -> Dict[str, Tensor]:
```

Comment:
Maybe I would prefer unroll inputs like: 
```suggestion
def _squad_compute(f1: Tensor, exact_match: Tensor, total: Tensor) -> Dict[str, Tensor]:
```

cc: @Borda